core/src/main/scala/dagr/core/execsystem/TaskTracker.scala

```diff
@@ -99,13 +93,29 @@ trait TaskTracker extends TaskManagerLike with LazyLogging {
         checkForCycles(task = task)
 
         // set the task id
-        val id = nextId
-        task._id = Some(id)
-        nextId += 1
+        val id = yieldAndThen(nextId)(nextId += 1)
```

Comment:
I tend to prefer `yieldAndThen(nextId) {nextId += 1}` braces for the second param list make it look more like a code block.  Up to you though.